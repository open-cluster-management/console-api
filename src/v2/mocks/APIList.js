/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

export const mockResponse = {
  body: {
    paths: [
      '/api',
      '/api/v1',
      '/apis',
      '/apis/',
      '/apis/admissionregistration.k8s.io',
      '/apis/admissionregistration.k8s.io/v1alpha1',
      '/apis/admissionregistration.k8s.io/v1beta1',
      '/apis/apiextensions.k8s.io',
      '/apis/apiextensions.k8s.io/v1beta1',
      '/apis/apiregistration.k8s.io',
      '/apis/apiregistration.k8s.io/v1',
      '/apis/apiregistration.k8s.io/v1beta1',
      '/apis/apps',
      '/apis/apps/v1',
      '/apis/apps/v1beta1',
      '/apis/apps/v1beta2',
      '/apis/authentication.k8s.io',
      '/apis/authentication.k8s.io/v1',
      '/apis/authentication.k8s.io/v1beta1',
      '/apis/authorization.k8s.io',
      '/apis/authorization.k8s.io/v1',
      '/apis/authorization.k8s.io/v1beta1',
      '/apis/autoscaling',
      '/apis/autoscaling/v1',
      '/apis/autoscaling/v2beta1',
      '/apis/batch',
      '/apis/batch/v1',
      '/apis/batch/v1beta1',
      '/apis/batch/v2alpha1',
      '/apis/certificates.k8s.io',
      '/apis/certificates.k8s.io/v1beta1',
      '/apis/certmanager.k8s.io',
      '/apis/certmanager.k8s.io/v1alpha1',
      '/apis/clusterregistry.k8s.io',
      '/apis/clusterregistry.k8s.io/v1alpha1',
      '/apis/compliance.mcm.ibm.com',
      '/apis/compliance.mcm.ibm.com/v1alpha1',
      '/apis/custom.metrics.k8s.io',
      '/apis/custom.metrics.k8s.io/v1beta1',
      '/apis/events.k8s.io',
      '/apis/events.k8s.io/v1beta1',
      '/apis/extensions',
      '/apis/extensions/v1beta1',
      '/apis/icp.ibm.com',
      '/apis/icp.ibm.com/v1',
      '/apis/mcm.ibm.com',
      '/apis/mcm.ibm.com/v1alpha1',
      '/apis/metrics.k8s.io',
      '/apis/metrics.k8s.io/v1beta1',
      '/apis/networking.k8s.io',
      '/apis/networking.k8s.io/v1',
      '/apis/policy',
      '/apis/policy.mcm.ibm.com',
      '/apis/policy.mcm.ibm.com/v1alpha1',
      '/apis/policy/v1beta1',
      '/apis/rbac.authorization.k8s.io',
      '/apis/rbac.authorization.k8s.io/v1',
      '/apis/rbac.authorization.k8s.io/v1beta1',
      '/apis/scheduling.k8s.io',
      '/apis/scheduling.k8s.io/v1beta1',
      '/apis/securityenforcement.admission.cloud.ibm.com',
      '/apis/securityenforcement.admission.cloud.ibm.com/v1beta1',
      '/apis/servicecatalog.k8s.io',
      '/apis/servicecatalog.k8s.io/v1beta1',
      '/apis/storage.k8s.io',
      '/apis/storage.k8s.io/v1',
      '/apis/storage.k8s.io/v1beta1',
      '/healthz',
      '/healthz/autoregister-completion',
      '/healthz/etcd',
      '/healthz/ping',
      '/healthz/poststarthook/apiservice-openapi-controller',
      '/healthz/poststarthook/apiservice-registration-controller',
      '/healthz/poststarthook/apiservice-status-available-controller',
      '/healthz/poststarthook/bootstrap-controller',
      '/healthz/poststarthook/ca-registration',
      '/healthz/poststarthook/generic-apiserver-start-informers',
      '/healthz/poststarthook/kube-apiserver-autoregistration',
      '/healthz/poststarthook/rbac/bootstrap-roles',
      '/healthz/poststarthook/scheduling/bootstrap-system-priority-classes',
      '/healthz/poststarthook/start-apiextensions-controllers',
      '/healthz/poststarthook/start-apiextensions-informers',
      '/healthz/poststarthook/start-kube-aggregator-informers',
      '/healthz/poststarthook/start-kube-apiserver-admission-initializer',
      '/healthz/poststarthook/start-kube-apiserver-informers',
      '/logs',
      '/metrics',
      '/openapi/v2',
      '/swagger-2.0.0.json',
      '/swagger-2.0.0.pb-v1',
      '/swagger-2.0.0.pb-v1.gz',
      '/swagger.json',
      '/swaggerapi',
      '/version',
    ],
  },
};

export const apiPath = {
  body: {
    kind: 'APIResourceList',
    apiVersion: 'v1',
    groupVersion: 'mcm.ibm.com/v1alpha1',
    resources: [
      {
        name: 'applicationrelationships',
        singularName: '',
        namespaced: true,
        kind: 'ApplicationRelationship',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
        shortNames: [
          'rel',
        ],
      },
      {
        name: 'applicationrelationships/status',
        singularName: '',
        namespaced: true,
        kind: 'ApplicationRelationship',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'applications',
        singularName: '',
        namespaced: true,
        kind: 'Application',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'channels',
        singularName: '',
        namespaced: true,
        kind: 'Channel',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'subscriptions',
        singularName: '',
        namespaced: true,
        kind: 'Subscription',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'placementrules',
        singularName: '',
        namespaced: true,
        kind: 'PlacementRule',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'applications/status',
        singularName: '',
        namespaced: true,
        kind: 'Application',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'clusterjoinrequests',
        singularName: '',
        namespaced: false,
        kind: 'ClusterJoinRequest',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'clusterjoinrequests/status',
        singularName: '',
        namespaced: false,
        kind: 'ClusterJoinRequest',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'clusterstatuses',
        singularName: '',
        namespaced: true,
        kind: 'ClusterStatus',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'clusterstatuses/topology',
        singularName: '',
        namespaced: true,
        kind: 'ClusterStatusTopology',
        verbs: [
          'create',
        ],
      },
      {
        name: 'deployableoverrides',
        singularName: '',
        namespaced: true,
        kind: 'DeployableOverride',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
        shortNames: [
          'do',
        ],
      },
      {
        name: 'deployableoverrides/status',
        singularName: '',
        namespaced: true,
        kind: 'DeployableOverride',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'deployables',
        singularName: '',
        namespaced: true,
        kind: 'Deployable',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'deployables/status',
        singularName: '',
        namespaced: true,
        kind: 'Deployable',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'placementbindings',
        singularName: '',
        namespaced: true,
        kind: 'PlacementBinding',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
        shortNames: [
          'pb',
        ],
      },
      {
        name: 'placementpolicies',
        singularName: '',
        namespaced: true,
        kind: 'PlacementPolicy',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
        shortNames: [
          'pp',
        ],
      },
      {
        name: 'placementrules',
        singularName: '',
        namespaced: true,
        kind: 'placementrule',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
        shortNames: [
          'pr',
        ],
      },
      {
        name: 'placementpolicies/status',
        singularName: '',
        namespaced: true,
        kind: 'PlacementPolicy',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'resourceviews',
        singularName: '',
        namespaced: true,
        kind: 'ResourceView',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'resourceviews/status',
        singularName: '',
        namespaced: true,
        kind: 'ResourceView',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'works',
        singularName: '',
        namespaced: true,
        kind: 'Work',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'works/result',
        singularName: '',
        namespaced: true,
        kind: 'ResourceViewResult',
        verbs: [
          'create',
        ],
      },
      {
        name: 'works/status',
        singularName: '',
        namespaced: true,
        kind: 'Work',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
      {
        name: 'worksets',
        singularName: '',
        namespaced: true,
        kind: 'WorkSet',
        verbs: [
          'create',
          'delete',
          'deletecollection',
          'get',
          'list',
          'patch',
          'update',
          'watch',
        ],
      },
      {
        name: 'worksets/status',
        singularName: '',
        namespaced: true,
        kind: 'WorkSet',
        verbs: [
          'get',
          'patch',
          'update',
        ],
      },
    ],
  },
};

