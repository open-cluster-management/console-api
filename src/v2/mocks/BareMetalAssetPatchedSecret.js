/** *****************************************************************************
 * Licensed Materials - Property of Red Hat, Inc.
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 ****************************************************************************** */

const mockResponse = {
  body: {
    apiVersion: 'v1',
    data: {
      password: 'dGVzdDE=',
      username: 'dGVzdDE=',
    },
    kind: 'Secret',
    metadata: {
      creationTimestamp: '2020-07-23T18:07:07Z',
      generateName: 'worker-0-bmc-secret',
      name: 'worker-0-bmc-secret',
      namespace: 'fake-cluster',
      ownerReferences: [{
        apiVersion: 'inventory.open-cluster-management.io/v1alpha1',
        kind: 'BareMetalAsset',
        name: 'worker-0',
        uid: '86683623-be4d-4210-9276-3fa320b45617',
      }],
      resourceVersion: '10449699',
      selfLink: '/api/v1/namespaces/fake-cluster/secrets/worker-0-bmc-secret-9bmgg',
      uid: '8c18d6a7-1f44-4928-81b5-489bdc4bacc7',
    },
    type: 'Opaque',
  },
};
export default mockResponse;
