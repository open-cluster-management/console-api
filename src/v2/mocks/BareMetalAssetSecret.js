/** *****************************************************************************
 * Licensed Materials - Property of Red Hat, Inc.
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 ****************************************************************************** */

const mockResponse = {
  body: {
    apiVersion: 'v1',
    data: {
      password: 'dGVzdDE=',
      username: 'dGVzdDE=',
    },
    kind: 'Secret',
    metadata: {
      creationTimestamp: '2020-07-23T18:07:07Z',
      generateName: 'worker-0-bmc-secret',
      name: 'worker-0-bmc-secret',
      namespace: 'fake-cluster',
      ownerReferences: [],
      resourceVersion: '10449699',
      selfLink: '/api/v1/namespaces/fake-cluster/secrets/worker-0-bmc-secret-9bmgg',
      uid: '8c18d6a7-1f44-4928-81b5-489bdc4bacc7',
    },
    type: 'Opaque',
  },
};
export default mockResponse;
