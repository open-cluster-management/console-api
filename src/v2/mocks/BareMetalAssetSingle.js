/** *****************************************************************************
 * Licensed Materials - Property of Red Hat, Inc.
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 ****************************************************************************** */

const mockResponse = {
  body: {
    apiVersion: 'inventory.open-cluster-management.io/v1alpha1',
    kind: 'BareMetalAsset',
    metadata: {
      annotations: {
        'kubectl.kubernetes.io/last-applied-configuration': '{"apiVersion":"inventory.open-cluster-management.io/v1alpha1","kind":"BareMetalAsset","metadata":{"annotations":{},"name":"baremetalasset-worker-0","namespace":"fake-cluster"},"spec":{"bmc":{"address":"ipmi://192.168.122.1:6233","credentialsName":"worker-0-bmc-secret"},"bootMACAddress":"00:1B:44:11:3A:B7","clusterDeployment":{"name":"cluster0","namespace":"cluster0"},"hardwareProfile":"hardwareProfile","role":"worker"}}\n',
      },
      creationTimestamp: '2020-03-05T12:01:11Z',
      generation: 1,
      name: 'baremetalasset-worker-0',
      namespace: 'fake-cluster',
      resourceVersion: '579175',
      selfLink: '/apis/inventory.open-cluster-management.io/v1alpha1/namespaces/fake-cluster/baremetalassets/baremetalasset-worker-0',
      uid: 'cb6eaf0e-2a60-4c4c-b068-2a4257fcce42',
    },
    spec: {
      bmc: {
        address: 'ipmi://192.168.122.1:6233',
        credentialsName: 'worker-0-bmc-secret',
      },
      bootMACAddress: '00:1B:44:11:3A:B7',
      clusterDeployment: {
        name: 'cluster0',
        namespace: 'cluster0',
      },
      hardwareProfile: 'hardwareProfile',
      role: 'worker',
    },
  },
};

export default mockResponse;
