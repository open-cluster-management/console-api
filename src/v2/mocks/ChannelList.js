// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
export const mockChannelsResponse = {
  body: {
    kind: 'ChannelList',
    metadata: {
      continue: '',
      resourceVersion: '6470654',
      selfLink: '/apis/apps.open-cluster-management.io/v1/channels',
    },
    apiVersion: 'apps.open-cluster-management.io/v1',
    items: [
      {
        apiVersion: 'apps.open-cluster-management.io/v1',
        kind: 'Channel',
        metadata: {
          annotations: {
            'open-cluster-management.io/user-group': 'c3lzdGVtOmNsdXN0ZXItYWRtaW5zLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'open-cluster-management.io/user-identity': 'a3ViZTphZG1pbg==',
          },
          creationTimestamp: '2020-10-22T14:47:17Z',
          generation: 1,
          name: 'ggithubcom-fxiang1-app-samples',
          namespace: 'ggithubcom-fxiang1-app-samples-ns',
          resourceVersion: '423984',
          selfLink: '/apis/apps.open-cluster-management.io/v1/namespaces/ggithubcom-fxiang1-app-samples-ns/channels/ggithubcom-fxiang1-app-samples',
          uid: 'af6dd93d-e545-4376-8d12-0c28277f0bd4',
        },
        spec: {
          pathname: 'https://github.com/fxiang1/app-samples.git',
          type: 'Git',
        },
      },
      {
        apiVersion: 'apps.open-cluster-management.io/v1',
        kind: 'Channel',
        metadata: {
          annotations: {
            'open-cluster-management.io/user-group': 'c3lzdGVtOmNsdXN0ZXItYWRtaW5zLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'open-cluster-management.io/user-identity': 'a3ViZTphZG1pbg==',
          },
          creationTimestamp: '2020-10-27T20:56:21Z',
          generation: 1,
          name: 'ggithubcom-kevinfcormier-sample-repo',
          namespace: 'ggithubcom-kevinfcormier-sample-repo-ns',
          resourceVersion: '5456091',
          selfLink: '/apis/apps.open-cluster-management.io/v1/namespaces/ggithubcom-kevinfcormier-sample-repo-ns/channels/ggithubcom-kevinfcormier-sample-repo',
          uid: '5d3ee038-b441-49ef-bf9a-30d30af13ab1',
        },
        spec: {
          pathname: 'https://github.com/KevinFCormier/sample-repo.git',
          secretRef: {
            name: 'ggithubcom-kevinfcormier-sample-repo-auth',
          },
          type: 'Git',
        },
      },
      {
        apiVersion: 'apps.open-cluster-management.io/v1',
        kind: 'Channel',
        metadata: {
          annotations: {
            'open-cluster-management.io/user-group': 'c3lzdGVtOmNsdXN0ZXItYWRtaW5zLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'open-cluster-management.io/user-identity': 'a3ViZTphZG1pbg==',
          },
          creationTimestamp: '2020-10-22T14:47:17Z',
          generation: 1,
          name: 'hkubernetes-chartsstoragegoogleapiscom',
          namespace: 'hkubernetes-chartsstoragegoogleapiscom-ns',
          resourceVersion: '423993',
          selfLink: '/apis/apps.open-cluster-management.io/v1/namespaces/hkubernetes-chartsstoragegoogleapiscom-ns/channels/hkubernetes-chartsstoragegoogleapiscom',
          uid: '592a18c7-8409-4708-a025-12fb58524812',
        },
        spec: {
          pathname: 'https://kubernetes-charts.storage.googleapis.com',
          type: 'HelmRepo',
        },
      },
      {
        apiVersion: 'apps.open-cluster-management.io/v1',
        kind: 'Channel',
        metadata: {
          annotations: {
            'open-cluster-management.io/user-group': 'c3lzdGVtOmNsdXN0ZXItYWRtaW5zLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'open-cluster-management.io/user-identity': 'a3ViZTphZG1pbg==',
          },
          creationTimestamp: '2020-10-27T16:20:49Z',
          generation: 1,
          name: 'oec2-dummy-dev1',
          namespace: 'oec2-dummy-dev1-ns',
          resourceVersion: '5262671',
          selfLink: '/apis/apps.open-cluster-management.io/v1/namespaces/oec2-dummy-dev1-ns/channels/oec2-dummy-dev1',
          uid: '7b41db1b-6dd2-4f0a-8235-0d9cb8688491',
        },
        spec: {
          pathname: 'http://ec2-dummy/dev1',
          type: 'ObjectBucket',
        },
      },
    ],
  },
};

export const mockChannelSecretResponse = {
  body: {
    kind: 'Secret',
    apiVersion: 'v1',
    metadata: {
      name: 'ggithubcom-kevinfcormier-sample-repo-auth',
      namespace: 'ggithubcom-kevinfcormier-sample-repo-ns',
      selfLink: '/api/v1/namespaces/ggithubcom-kevinfcormier-sample-repo-ns/secrets/ggithubcom-kevinfcormier-sample-repo-auth',
      uid: 'e10bb428-324b-46a8-9d2b-3a84bb64ee84',
      resourceVersion: '5492077',
      creationTimestamp: '2020-10-27T13:21:52Z',
      labels: {
        'apps.open-cluster-management.io/serving-channel': 'true',
      },
      annotations: {
        'apps.open-cluster-management.io/serving-channel': 'ggithubcom-kevinfcormier-sample-repo-ns/ggithubcom-kevinfcormier-sample-repo',
      },
    },
    data: {
      accessToken: 'c29tZS1mYWtlLWFjY2Vzcy10b2tlbg==',
      user: 'ZmFrZXVzZXI=',
    },
    type: 'Opaque',
  },
};
