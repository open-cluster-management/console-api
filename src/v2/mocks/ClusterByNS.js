/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */

export const kubeSystem = {
  body: {
    kind: 'ClusterList',
    apiVersion: 'clusterregistry.k8s.io/v1alpha1',
    metadata: {
      selfLink: '/apis/clusterregistry.k8s.io/v1alpha1/clusters',
      resourceVersion: '136667',
    },
    items: [
      {
        kind: 'Cluster',
        apiVersion: 'clusterregistry.k8s.io/v1alpha1',
        metadata: {
          name: 'hub-cluster',
          namespace: 'kube-system',
          selfLink: '/apis/clusterregistry.k8s.io/v1alpha1/namespaces/kube-system/clusters/hub-cluster',
          uid: 'd9f3a5aa-9f19-11e8-855e-f2b998610544',
          resourceVersion: '136558',
          creationTimestamp: '2018-08-13T16:56:41Z',
          labels: {
            cloud: 'IBM',
            datacenter: 'raleigh',
            environment: 'Dev',
            name: 'hub-cluster',
            owner: 'development',
            region: 'US',
            vendor: 'ICP',
          },
          annotations: {
            'mcm.ibm.com/user-group': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'mcm.ibm.com/user-identity': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=',
          },
          finalizers: [
            'finalizer.hcm.ibm.com',
          ],
        },
        spec: {
          kubernetesApiEndpoints: {
            serverEndpoints: [
              {
                serverAddress: '9.42.80.212:8001',
              },
            ],
          },
          authInfo: {},
        },
        status: {
          conditions: [
            {
              type: 'OK',
              status: '',
              lastHeartbeatTime: '2018-08-15T19:41:20Z',
              lastTransitionTime: null,
            },
          ],
        },
      },
      {
        kind: 'Cluster',
        apiVersion: 'clusterregistry.k8s.io/v1alpha1',
        metadata: {
          name: 'new-cluster',
          namespace: 'kube-system',
          selfLink: '/apis/clusterregistry.k8s.io/v1alpha1/namespaces/kube-system/clusters/new-cluster',
          uid: 'd9f3a6aa-9f19-11f8-855e-f2b908610543',
          resourceVersion: '137558',
          creationTimestamp: '2018-08-13T16:56:41Z',
          labels: {
            cloud: 'IBM',
            datacenter: 'raleigh',
            environment: 'Dev',
            name: 'new-cluster',
            owner: 'development',
            region: 'US',
            vendor: 'ICP',
          },
          annotations: {
            'mcm.ibm.com/user-group': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'mcm.ibm.com/user-identity': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=',
          },
          finalizers: [
            'finalizer.hcm.ibm.com',
          ],
        },
        spec: {
          kubernetesApiEndpoints: {
            serverEndpoints: [
              {
                serverAddress: '9.42.80.212:8001',
              },
            ],
          },
          authInfo: {},
        },
        status: {
          conditions: [
            {
              type: '',
              status: '',
              lastHeartbeatTime: '2018-08-15T19:41:20Z',
              lastTransitionTime: null,
            },
          ],
        },
      },
    ],
  },
};

export default {
  body: {
    kind: 'ClusterList',
    apiVersion: 'clusterregistry.k8s.io/v1alpha1',
    metadata: {
      selfLink: '/apis/clusterregistry.k8s.io/v1alpha1/clusters',
      resourceVersion: '136667',
    },
    items: [
      {
        kind: 'Cluster',
        apiVersion: 'clusterregistry.k8s.io/v1alpha1',
        metadata: {
          name: 'managed-cluster',
          namespace: 'default',
          selfLink: '/apis/clusterregistry.k8s.io/v1alpha1/namespaces/default/clusters/mycluster.icp',
          uid: 'd9f3a5aa-9f19-11e8-855e-f2b998610544',
          resourceVersion: '136558',
          creationTimestamp: '2018-08-13T16:56:41Z',
          labels: {
            cloud: 'IBM',
            datacenter: 'toronto',
            environment: 'Dev',
            name: 'managed-cluster',
            owner: 'marketing',
            region: 'US',
            vendor: 'ICP',
          },
          annotations: {
            'mcm.ibm.com/user-group': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'mcm.ibm.com/user-identity': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=',
          },
          finalizers: [
            'finalizer.hcm.ibm.com',
          ],
        },
        spec: {
          kubernetesApiEndpoints: {
            serverEndpoints: [
              {
                serverAddress: '9.42.80.212:8001',
              },
            ],
          },
          authInfo: {},
        },
        status: {},
      },
    ],
  },
};
