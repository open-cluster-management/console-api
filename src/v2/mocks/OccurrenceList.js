/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2020 Red Hat, Inc.

/* eslint-disable import/prefer-default-export */
export const mockOccurrences = {
  data: {
    occurrences: [
      {
        name: 'id-mycluster-account/providers/securityAdvisorNewProviderId/occurrences/securityAdvisorOccurrenceId',
        noteName: 'id-mycluster-account/providers/securityAdvisorNewProviderId/notes/securityAdvisorNoteID',
        updateTime: '2019-07-10T14:07:46.596382Z',
        createTime: '2019-07-10T14:07:46.596357Z',
        shortDescription: 'Registered services1',
        context: {
          accountId: 'id-mycluster-account',
          region: 'location',
          resourceType: 'worker',
          resourceName: 'www.myapp.com',
          resourceId: 'pluginId',
          resourceCrn: null,
          serviceName: 'application',
          serviceCrn: null,
        },
        reportedBy: {
          id: 'securityAdvisorNewProviderId',
          title: 'Title for Note',
          url: null,
        },
        finding: {
          severity: 'HIGH',
          certainty: 'HIGH',
          networkConnection: null,
          nextSteps: [
            {
              title: 'title here',
              url: 'Details URL',
            },
          ],
          dataTransferred: null,
        },
        securityClassification: {
          securityCategories: [
            'System and communications protections',
          ],
        },
      },
    ],
  },
};
