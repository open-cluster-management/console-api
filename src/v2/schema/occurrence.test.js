/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import supertest from 'supertest';
import nock from 'nock';
import server, { GRAPHQL_PATH } from '../index';
import { mockOccurrences } from '../mocks/OccurrenceList';

describe('Occurrences Resolver', () => {
  beforeAll(() => {
    // specify the url to be intercepted
    const APIServer = nock('http://0.0.0.0/findings');

    // define the method to be intercepted
    APIServer.post('/v1/id-mycluster-account/graph', /\.*occurrences\.*/gi)
      .reply(200, mockOccurrences);
  });

  test('Correctly Resolves Occurrences Query', () => new Promise((done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          occurrences {
            name
            noteName
            updateTime
            createTime
            shortDescription
            context
            reportedBy
            finding
            securityClassification
          }
        }
      `,
      })
      .end((err, res) => {
        const textMessage = JSON.parse(res.text);
        expect(textMessage).toMatchSnapshot();
        done();
      });
  }));
});
