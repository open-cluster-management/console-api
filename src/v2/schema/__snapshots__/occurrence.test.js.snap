// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Occurrences Resolver Correctly Resolves Occurrences Query 1`] = `
Object {
  "data": Object {
    "occurrences": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Cannot read property 'errors' of undefined",
      "path": Array [
        "occurrences",
      ],
    },
  ],
}
`;
